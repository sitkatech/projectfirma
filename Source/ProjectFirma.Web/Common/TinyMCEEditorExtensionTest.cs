/*-----------------------------------------------------------------------
<copyright file="TinyMCEEditorExtensionTest.cs" company="Tahoe Regional Planning Agency and Environmental Science Associates">
Copyright (c) Tahoe Regional Planning Agency and Environmental Science Associates. All rights reserved.
<author>Environmental Science Associates</author>
</copyright>

<license>
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License <http://www.gnu.org/licenses/> for more details.

Source code is available upon request via <support@sitkatech.com>.
</license>
-----------------------------------------------------------------------*/
using ApprovalTests;
using ApprovalTests.Reporters;
using ProjectFirma.Web.Controllers;
using LtInfo.Common;
using LtInfo.Common.Mvc;
using NUnit.Framework;

namespace ProjectFirma.Web.Common
{
    [TestFixture]
    public class TinyMCEEditorExtensionTest
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithMinimalToolbarTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.Minimal,
                 false, null);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithMinimalWithImagesToolbarTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.MinimalWithImages,
                 false, null);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithNoToolbarTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.None,
                 false, null);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithAllToolbarsTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.All,
                 false, null);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithAllOnOneRowToolbarsTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.AllOnOneRow,
                 false, null);
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateJavascriptWithAllOnOneRowNoMaximizeToolbarsTest()
        {
            const string modelID = "ProgramPageContent";
            var result = TinyMCEExtension.GenerateJavascript(modelID,
                TinyMCEExtension.TinyMCEToolbarStyle.AllOnOneRowNoMaximize,
                 false, null);
            Approvals.Verify(result);
        }

    }
}
