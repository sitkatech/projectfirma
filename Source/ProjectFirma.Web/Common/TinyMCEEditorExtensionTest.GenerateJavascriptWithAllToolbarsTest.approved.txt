<script language="javascript" type="text/javascript">
                // <![CDATA[
                jQuery(document).ready(function ()
                {
                   tinymce.init({
                            selector: '#TinyMCEEditorForProgramPageContent',
                            menubar: false,
                            toolbar: 'code | styleselect | bold italic removeformat | bullist numlist outdent indent | image table hr charmap | link unlink anchor | styles | fontfamily | fullscreen ',
                            entity_encoding: 'named+numeric',
                            plugins: 'code lists link image table code help wordcount charmap anchor fullscreen',
                            toolbar_mode: 'wrap',
                            browser_spellcheck: true,
                            file_picker_types: 'image',
                            images_file_types: 'jpg,svg,webp,gif',
                            image_title: true,
                            file_picker_callback: (cb, value, meta) => {
                              const input = document.createElement("input")
                              input.setAttribute("type", "file")
                              input.setAttribute("accept", "image/*")
                              input.addEventListener("change", e => {
                                const file = e.target.files[0];

                                const reader = new FileReader();
                                reader.addEventListener("load", () => {
                                  const id = "blobid" + new Date().getTime();
                                  const blobCache = tinymce.activeEditor.editorUpload.blobCache;
                                  const base64 = reader.result.split(",")[1];
                                  const blobInfo = blobCache.create(id, file, base64);
                                  blobCache.add(blobInfo);

                                  /* call the callback and populate the Title field with the file name */
                                  cb(blobInfo.blobUri(), { title: file.name });
                                });
                                reader.readAsDataURL(file)
                              })
                              input.click()
                            },
                            setup: function (editor) {
                                editor.on('FullscreenStateChanged', function (e) {
                                    if (e.state) {
                                        $('.modal-dialog').attr('style', 'transform: none !important');
                                    } else {
                                        $('.modal-dialog').attr('style', 'transform: translate(0,0)');
                                    }
                                });
                            }
                    });
                });
                // ]]>
            </script>